_posts/2025-08-18-opensearch-mcp-server-empowering-ec2-teams-droplet-analysis.md

```diff
@@ -0,0 +1,328 @@
+---
+
+# Introduction
+
+Amazon EC2 team recently leveraged OpenSearch's Model Context Protocol (MCP) server to revolutionize their droplet quality investigations. By connecting Amazon Q with multiple OpenSearch clusters simultaneously, they've transformed a previously manual, time-consuming process into an efficient, automated workflow. This blog explores how the EC2 team implemented this solution, the technical setup involved, and the impressive results they've achieved.
+
+# The Challenge
+
+The EC2 team manages droplet quality across their fleet, investigating failures and impairments to maintain reliability. Their data was distributed across multiple OpenSearch clusters in various regions, each storing logs for different components. Engineers faced a multi-step process when troubleshooting:
+
+* Search in one cluster to find initial information
+* Use those findings to query another cluster
+* Correlate data across multiple sources
+* Repeat until root cause is identified
+
+This manual process required deep knowledge of data locations and search patterns. The team realized their engineers' time would be better spent solving problems rather than hunting for and correlating data across systems.
+
+# The Solution
+
+The EC2 team implemented a solution combining:
+* **Amazon Q CLI** as the intelligent agent to analyze data and provide insights
+
+> **Note:** While this blog demonstrates the solution with Amazon Q CLI, you can replace it with any other agentic solution such as Claude, OpenAI, or other AI agents that support the Model Context Protocol
+* **OpenSearch MCP Server** as the connectivity layer to multiple OpenSearch clusters
+* **EC2 instance** running the agent with an attached instance role for secure cluster access
+* **IAM-based Authentication** for secure authentication to different clusters
+
+![OpenSearch MCP Server Architecture](/assets/media/blog-images/2025-08-18-opensearch-mcp-server-empowering-ec2-teams-droplet-analysis/mcp-architecture-diagram.png){: .img-fluid}
+
+The key innovation was leveraging OpenSearch MCP Server's multi-cluster connectivity feature, which allowed them to interact with multiple OpenSearch clusters through a single interface, avoiding tool proliferation that could overwhelm the agent.
+
+# Implementation Steps
+
+## Step 1: Set Up IAM Authentication
+
+The EC2 team's infrastructure spans three OpenSearch clusters across different AWS accounts:
+
+1. **hardware-logs**: Stores hardware engineering data and failure events
+2. **droplet-metrics**: Contains performance metrics and system telemetry  
+3. **system-events**: Holds lifecycle events and administrative actions
+
+To securely access these distributed clusters, the team implemented an EC2 instance-based authentication solution that allows them to connect to all three clusters from a single location while maintaining proper security controls.
+
+### IAM Setup Steps:
+
+#### 1.1 Create the Three Cluster Access Roles (in Cluster Accounts)
```

Comment:
```suggestion
1. Create the three cluster access roles (in cluster accounts): First, the EC2 team created the three IAM roles, in their respective AWS accounts, that would have access to each OpenSearch cluster. Each role needed a trust policy that would allow the EC2 instance role from the main account to assume it.
```